File: backups/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/brackets/PairedBracket.java

package io.github.abc15018045126.sora.lang.brackets;

/**
 * Describes paired brackets
 *
 * @author abc15018045126
 */
public class PairedBracket {

    public final int leftIndex, leftLength, rightIndex, rightLength;

    /**
     * Currently length is always 1.
     *
     * @see #PairedBracket(int, int, int, int)
     */
    public PairedBracket(int leftIndex, int rightIndex) {
        this(leftIndex, 1, rightIndex, 1);
    }

    /**
     * @param leftIndex   Index of left bracket in text
     * @param leftLength  Text length of left bracket
     * @param rightIndex  Index of right bracket in text
     * @param rightLength Text length of right bracket
     */
    public PairedBracket(int leftIndex, int leftLength, int rightIndex, int rightLength) {
        this.leftIndex = leftIndex;
        this.leftLength = leftLength;
        this.rightIndex = rightIndex;
        this.rightLength = rightLength;
    }
}

File: android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/brackets/PairedBracket.kt
package io.github.abc15018045126.sora.lang.brackets


class PairedBracket @JvmOverloads constructor(
    @JvmField val leftIndex: Int,
    @JvmField val leftLength: Int = 1,
    @JvmField val rightIndex: Int,
    @JvmField val rightLength: Int = 1
)
